iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -1030,6 +1037,52 @@ def index(self, value: dict[IndexSymbol, IndexSequence]):
         CustomOp.index.fset(self, value)
 
 
+@define_op("extract")
+@dataclass
+class Extract(CustomOp):
+    """
+    Op Rationale:
+
+    Extract is an op used to represent extracting of
+    a scalar from TKW's 1-D vector on the specified index.
+
+    This can also be viewed as indexing/slicing on the fastest
+    dimension. Hence, the semantic of this op is designed to
+    see itself as a reduction on the indexed/fastest dimension.
+    """
+
+    register_: fx.Proxy
+    offset: IndexExpr | int
```

Comment:
In numpy semantics, if a = [4x8], then a[0] gives you a [8] tensor. This makes sense. What if a is more than 2d? If we don't want to handle that, we should just assert.

Reply:
okay let me reread, with the new information I have. Ignore this.